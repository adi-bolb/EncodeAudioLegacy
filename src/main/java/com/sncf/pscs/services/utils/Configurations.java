package com.sncf.pscs.services.utils;

public interface Configurations {
	
	// Nom du fichier de configuration pour log4j.
	public final String LOG4J_FILENAME = "pscsEsb-log4j.xml";
	
	// Connection factory pour faire de la publication.
	public static final String MOM_PUB_CONNECTION_FACTORY_NAME = "jms/com/sncf/pscs/topicFactory";
	
	public static final String FABRIQUE_DE_CONNECTIONS = "jms/com/sncf/pscs/factoryPool";
	public static final String FILE_DESTINATION = "jms/com/sncf/pscs/failedQueue";
	
	public static final String messageFormatFaultName = "messageFormatFault"; 
	public static final String ERREUR_JMS_STACK = "probl�me avec la pile JMS. ";
	public static final String ERREUR_JMS_NAMING = "probl�me avec l'acces aux ressources JMS. ";
	public static final String ERREUR_JMS_FORMAT = "Format du message JMS  non reconnu : \n";
	public static final String ERREUR_COM_JMS = "Erreur de communication avec la file de sauvegarde ";
	
	// type enum pour distinguer le type de files d'attentes 
	public enum typeDestination {
		INFOVOYAGEUR,
		MAINTENANCE		
	}

	
	/** 
	 * Divers
	 */
	public static final String BLANK = "";
	public static final String SEPARATOR = ";";
	
	
	/**
	 * Nom des champs des tables de configuration de la BDD
	 */
	public static final String ENCODING_ACTIVATE = "encoding_activate";
	public static final String PATH_PROGRAM_EXE = "path_program_exe";
	public static final String FINAL_AUDIO_FILE_EXTENSION = "final_audio_file_extension";
	public static final String ENCODING_BITRATE = "encoding_bitrate";
	public static final String ENCODING_VBR_QUALITY = "encoding_vbr_quality";
	public static final String PATH_UPLOAD_CCUE_FILE = "path_upload_ccue_file";
	public static final String AUDIO_TEMP_PATH = "audio_temp_path";
	public static final String ANNONCE_AUDIO_TEST_FILE = "annonce_audio_test_file";
	public static final String MESSAGE_ANNONCE_AUDIO_TEST = "message_annonce_audio_test";
	public static final String URL_LOCAL_SERVER_POST = "url_local_server_post";
	public static final String URL_LOCAL_SERVER_GET = "url_local_server_get";
	public static final String URL_EMBARQUE_SERVER_GET = "url_embarque_server_get";
	public static final String URL_EMBARQUE_SERVER_POST = "url_embarque_server_post";
	public static final String SSL_TRUST_STORE_PATH = "ssl_trust_store_path";
	public static final String SSL_TRUST_STORE_PASSWD = "ssl_trust_store_passwd";
	
	
}
