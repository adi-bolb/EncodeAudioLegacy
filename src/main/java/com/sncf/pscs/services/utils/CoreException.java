package com.sncf.pscs.services.utils;

/**
 * @author PAG
 *
 */
public class CoreException extends Exception{

	private static final long serialVersionUID = 563728075825478349L;

	/**
	 * @param message
	 */
	public CoreException (final String message) {
		super(message);
	}

	/**
	 * @param exeption
	 */
	public CoreException (final Exception exeption) {
		super(exeption);
	}

	/**
	 * @param message
	 * @param exeption
	 */
	public CoreException (final String message, final Exception exeption) {
		super(message, exeption);
	}
	
}
