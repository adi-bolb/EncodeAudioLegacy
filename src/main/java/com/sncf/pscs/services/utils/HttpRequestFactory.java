package com.sncf.pscs.services.utils;

import java.io.BufferedReader;
import java.io.ByteArrayOutputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.net.MalformedURLException;
import java.net.URL;
import java.net.URLConnection;
import java.security.Security;


/**
 * @author SEB Classe factory permettant de lancer des actions/verbes HTTP(s)
 */
public class HttpRequestFactory {
	
	// =======================================
	// Variables static

	//private static final Logger LOG = Logger.getLogger(HttpRequestFactory.class);
	private static final String LINE_END = "\r\n";
	private static final String TWO_HYPHENS = "--";
	private static final String BOUNDARY = "*****";

	
	// =======================================
	// Methodes

	/**
	 * Post un fichier pour le d�poser sur un serveur HTTPS
	 * 
	 * @param sslTrustStore
	 * @param sslTrustStorePassword
	 * @param sUrl
	 * @param decoded
	 * @param fileName
	 * @throws CoreException 
	 */
	public static void postFileToHttpsServer(String sslTrustStore, String sslTrustStorePassword, String sUrl, byte[] decoded, String fileName) throws CoreException  {
		try{
			DataOutputStream dos = null;
			
			Security.setProperty("ssl.SocketFactory.provider", "com.ibm.jsse2.SSLSocketFactoryImpl");
			Security.setProperty("ssl.ServerSocketFactory.provider", "com.ibm.jsse2.SSLServerSocketFactoryImpl");
			
			System.setProperty("javax.net.ssl.trustStore", sslTrustStore);
			System.setProperty("javax.net.ssl.trustStorePassword", sslTrustStorePassword);

//			javax.net.ssl.HttpsURLConnection connection =(javax.net.ssl.HttpsURLConnection) new URL(sUrl).openConnection();
//			connection.setHostnameVerifier(new HostnameVerifier(){
//					public boolean verify(String urlHostName, SSLSession session) {
//						return true;
//					}
//				});
			URL url = new URL(sUrl);
			URLConnection connection = url.openConnection();
			connection.setDoOutput(true);
			connection.setAllowUserInteraction(false);
			// Allow Inputs
			connection.setDoInput(true);
			// Allow Outputs
			connection.setDoOutput(true);
			// Don't use a cached copy.
			connection.setUseCaches(false);
			// Use a post method.
			connection.setRequestProperty("Request-Method", "POST");
			connection.setRequestProperty("Connection", "Keep-Alive");
			connection.setRequestProperty("Content-Type", "multipart/form-data;boundary=" + BOUNDARY);
			dos = new DataOutputStream(connection.getOutputStream());

			dos.writeBytes(TWO_HYPHENS + BOUNDARY + LINE_END);
			dos.writeBytes("Content-Disposition: form-data; name=\"FILE1\";" + " filename=\"" + fileName + "\"" + LINE_END);
			dos.writeBytes(LINE_END);
			dos.write(decoded);
			// send multipart form data necesssary after file data...
			dos.writeBytes(LINE_END);
			dos.writeBytes(TWO_HYPHENS + BOUNDARY + TWO_HYPHENS + LINE_END);

			// close streams
			dos.flush();
			dos.close();
			// ------------------ read the SERVER RESPONSE
			readServerResponse(connection);
		}catch (Exception ex){
			throw new CoreException("Probleme avec le serveur HTTP: " + ex.getMessage());
		}
			
	}

	/**
	 * Read the server response 
	 * @param conn
	 * @throws CoreException 
	 */
	public static void readServerResponse(URLConnection conn) throws CoreException{
		
		try {
			BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(conn.getInputStream()));
			String str;
//			//TODO traiter les messages d'erreur HTTP
			while ((str = bufferedReader.readLine()) != null) {
//				//if (LOG.isDebugEnabled()){
//				//	LOG.debug(str + "");
//				//}
			}
			
			bufferedReader.close();

		} catch (IOException ioex) {
			throw new CoreException("Probleme avec la reponse du serveur HTTP: " + ioex.getMessage());
		} 
	}
	
	/**
	 * Post un fichier pour le d�poser sur un serveur HTTPS
	 * 
	 * @param sUrl
	 * @return byte[]
	 * @throws CoreException 
	 */
	public static byte[] getFileFromHttpServer(String sUrl) throws CoreException{
		
		//url = http://10.20.2.178/audio/
		//fileName = ccue.mp3
		try {
			URL url;
			url = new URL(sUrl);
			URLConnection connection = url.openConnection();
			connection.setDoOutput(true);
			connection.setAllowUserInteraction(false);
			// Allow Inputs
			connection.setDoInput(true);
			// Allow Outputs
			connection.setDoOutput(true);
			// Don't use a cached copy.
			connection.setUseCaches(false);
			
			// Use a GET method.
			connection.setRequestProperty("Request-Method", "GET");
			connection.setRequestProperty("Connection", "Keep-Alive");
						
			// ------------------ read the SERVER RESPONSE
			InputStream in = connection.getInputStream();
			ByteArrayOutputStream bos = new ByteArrayOutputStream();
					
							
			byte[]buff = new byte[1];
			int l = in.read(buff);
			//StringBuffer sb=new StringBuffer();
			
			while(l>0)
			{		
			   // pas d'encodage / decodage ; rapatriement du fichier binaire audio 
			   bos.write(buff);
			   l = in.read(buff);		   
			}
//			String s=sb.toString();	
			return bos.toByteArray();
		} catch (MalformedURLException e) {
			throw new CoreException("Erreur lors de la r�cup�ration du fichier a l'adresse : " + sUrl,e);
		} catch (IOException e) {
			throw new CoreException("Erreur lors de la r�cup�ration du fichier a l'adresse : " + sUrl,e);
		}
	}
}
