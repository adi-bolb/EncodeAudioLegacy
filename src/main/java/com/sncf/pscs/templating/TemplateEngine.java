package com.sncf.pscs.templating;

import java.util.Map;

public interface TemplateEngine {

	String compile(String template, Map<String, String> vals);

}
