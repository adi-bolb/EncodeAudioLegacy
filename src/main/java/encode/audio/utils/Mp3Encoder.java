package encode.audio.utils;

import java.io.IOException;

public class Mp3Encoder {

	public static int launchMp3Exec(String cmd) throws IOException,
			InterruptedException {
		Process p1 = Runtime.getRuntime().exec(cmd);
		p1.getOutputStream().close();
		p1.getInputStream().close();
		p1.getErrorStream().close();
		p1.waitFor();
	
		// Exï¿½cution de la commande d'encodage audio
		//LOG.info("Fin de l'encodage du fichier audio, code retour de la commande :" + p.exitValue());
		int exitValue = p1.exitValue();
		return exitValue;
	}

}
