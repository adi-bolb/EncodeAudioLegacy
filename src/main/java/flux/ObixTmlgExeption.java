package flux;

public class ObixTmlgExeption extends Exception {

}
