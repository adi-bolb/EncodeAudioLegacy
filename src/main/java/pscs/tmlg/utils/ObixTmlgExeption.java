package pscs.tmlg.utils;

public class ObixTmlgExeption extends Exception {

}
