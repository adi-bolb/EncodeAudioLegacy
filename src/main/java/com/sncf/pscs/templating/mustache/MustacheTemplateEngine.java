package com.sncf.pscs.templating.mustache;

import java.io.IOException;
import java.io.StringWriter;
import java.util.Map;

import com.github.mustachejava.DefaultMustacheFactory;
import com.github.mustachejava.Mustache;
import com.sncf.pscs.templating.TemplateEngine;

public class MustacheTemplateEngine implements TemplateEngine {

	public String compile(String template, Map<String, String> vals) {
		Mustache mustache = new DefaultMustacheFactory().compile(template);
	    StringWriter writer = new StringWriter();
		try {
			mustache.execute(writer, vals ).flush();
		} catch (IOException e) {
			throw new RuntimeException(e);
		}
		return writer.toString();
	}

}
