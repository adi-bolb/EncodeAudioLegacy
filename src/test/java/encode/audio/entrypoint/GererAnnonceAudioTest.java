package encode.audio.entrypoint;

import static org.mockito.Matchers.anyString;

import java.io.File;
import java.io.IOException;
import java.io.InputStream;
import java.util.HashMap;
import java.util.Map;

import org.approvaltests.legacycode.LegacyApprovals;
import org.approvaltests.reporters.UseReporter;
import org.junit.Test;
import org.junit.rules.TemporaryFolder;
import org.junit.runner.RunWith;
import org.mockito.Mockito;
import org.mockito.invocation.InvocationOnMock;
import org.mockito.stubbing.Answer;
import org.powermock.api.mockito.PowerMockito;
import org.powermock.core.classloader.annotations.PrepareForTest;
import org.powermock.modules.junit4.PowerMockRunner;

import templating.TemplateEngine;
import templating.mustache.MustacheTemplateEngine;
import encode.audio.utils.CoreException;
import encode.audio.utils.Mp3Encoder;
import flux.ObixTmlgExeption;

@RunWith(PowerMockRunner.class)
@PrepareForTest(Mp3Encoder.class)
public class GererAnnonceAudioTest {

	private static final String TEST_MP3_FILENAME = "test.mp3";
	private static final Object[] ALL_BOOLEAN_VALUES = { true, false };

	// @Test
	// public void runOnce() throws Exception {
	// assertEquals("kbaskhjfksadjf", allGererAnnonceAudio(true, "wav", 128,
	// "lame"));
	// }

	@Test
	public void downLoadError_transformationLock() throws Exception {
		Object[] encodingActivated = ALL_BOOLEAN_VALUES;

		Object[] conversionBinaryName = { "lame" };
		Object[] bitrate = { 128 };

		Object[] audioExtension = { "mp3", "wav" };
		Object[] encoding = ALL_BOOLEAN_VALUES;

		LegacyApprovals.LockDown(this,
				"downloadError_GererAnnonceAudioCatchExceptions",
				encodingActivated, audioExtension, bitrate,
				conversionBinaryName, encoding);
	}

	@Test
	public void transformationLock() throws Exception {
		Object[] encodingActivated = ALL_BOOLEAN_VALUES;

		Object[] conversionBinaryName = { "lame" };
		Object[] bitrate = { 128 };
		Object[] audioExtension = { "mp3", "wav" };
		Object[] encoding = ALL_BOOLEAN_VALUES;
		Object[] destAudioFileUC = { new UC_DestAudioFile_NotAlreadyEncoded(),
				new UC_DestAudioFile_AlreadyEncoded(TEST_MP3_FILENAME) };

		LegacyApprovals.LockDown(this,
				"default_GererAnnonceAudioCatchExceptions", encodingActivated,
				audioExtension, bitrate, conversionBinaryName, encoding,
				destAudioFileUC);
	}

	@Test(expected = AppTechnicalException.class)
	public void cannot_publish_target_audio_file_readfileissue()
			throws Exception {
		Boolean encodingActivated = Boolean.TRUE;
		String audioExtension = "mp3";
		Integer bitrate = new Integer(128);
		String conversionBinaryName = "lame";
		Boolean encodingSuccess = Boolean.TRUE;
		UC_DestAudioFile destAudioFile = new UC_DestAudioFile_AlreadyEncoded(
				TEST_MP3_FILENAME);

		TemporaryFolder rootFolder = setup();
		final File downloadDir = rootFolder.newFolder();
		GererAnnonceAudio aa = gererAnnonceAudio_WithNoLocalServerFolder();
		gererAnnonceAudio_simulate_readissue(encodingActivated, audioExtension, bitrate,
				conversionBinaryName, encodingSuccess, aa, downloadDir,
				destAudioFile);
	}

	@Test(expected = AppTechnicalException.class)
	public void cannot_publish_target_audio_file_writefileissue()
			throws Exception {
		Boolean encodingActivated = Boolean.TRUE;
		String audioExtension = "mp3";
		Integer bitrate = new Integer(128);
		String conversionBinaryName = "lame";
		Boolean encodingSuccess = Boolean.TRUE;
		UC_DestAudioFile destAudioFile = new UC_DestAudioFile_AlreadyEncoded(
				TEST_MP3_FILENAME);

		TemporaryFolder rootFolder = setup();
		final File downloadDir = rootFolder.newFolder();
		GererAnnonceAudio aa = gererAnnonceAudio_WithNoLocalServerFolder();
		allGererAnnonceAudio(encodingActivated,
				audioExtension, bitrate, conversionBinaryName, encodingSuccess,
				aa, downloadDir, destAudioFile);

	}

	private String gererAnnonceAudio_simulate_readissue(
			Boolean encodingActivated, String audioExtension, Integer bitrate,
			String conversionBinaryName, Boolean encodingSuccess,
			GererAnnonceAudio aa, File downloadDir,
			UC_DestAudioFile destAudioFile) throws Exception {

		downloadDir.mkdir();
		HttpDataObj httpDataObj = Mockito.mock(HttpDataObj.class);
		Mockito.when(httpDataObj.getString("audio_temp_path"))
				.thenReturn(downloadDir.getAbsolutePath() + "/").thenReturn("writeissue");
		Mockito.when(httpDataObj.getString("url_embarque_server_get"))
				.thenReturn("http://localhost/get");

		DataObject configAudioTmp = new AudioDataObject(encodingActivated,
				audioExtension); // url_embarque_server_get,

		final String destinationFullPath = downloadDir.getAbsolutePath() + "/"
				+ "10.151.156.180Mon_Nov_04_140724_CET_2013343.mp3";
		final String resourceName = TEST_MP3_FILENAME;

		Answer<Integer> encodingOperation = encodingSuccess ? successfullEncoding(
				resourceName, destinationFullPath) : encodingFailure();

		PowerMockito.mockStatic(Mp3Encoder.class);
		PowerMockito.when(Mp3Encoder.launchMp3Exec(anyString())).thenAnswer(
				encodingOperation);

		destAudioFile.activateUC(destinationFullPath);

		// When
		String flux = aa.demanderFlux(buildXmlRequest(audioExtension),
				configAudioTmp, httpDataObj);

		// Then
		return flux;
	}

	private GererAnnonceAudio gererAnnonceAudio_WithNoLocalServerFolder()
			throws IOException {
		GererAnnonceAudio aa = new GererAnnonceAudio() {
			@Override
			public byte[] downloadAudioFileFromHttpServer(String fileUrl)
					throws CoreException {
				try {
					InputStream resource = getClass()
							.getClassLoader()
							.getResourceAsStream(
									"10.151.156.180Mon_Nov_04_140724_CET_2013343.wav");
					byte[] bts = new byte[1000000];
					resource.read(bts);
					return bts;
				} catch (Exception e) {
					throw new CoreException(e);
				}
			}

			@Override
			protected String getLocalServerFolder() {

				return "/no/local/server/folder";
			}
		};
		return aa;
	}

	public String downloadError_GererAnnonceAudioCatchExceptions(
			Boolean encodingActivated, String audioExtension, Integer bitrate,
			String conversionBinaryName, Boolean encodingSuccess) {
		try {
			TemporaryFolder rootFolder = setup();
			try {
				GererAnnonceAudio aa;
				aa = gererAnnonceAudio_DownloadError();

				final File downloadDir = rootFolder.newFolder();
				return downloadError_GererAnnonceAudio(encodingActivated,
						audioExtension, bitrate, conversionBinaryName,
						encodingSuccess, aa, downloadDir);
			} finally {
				teardown(rootFolder);

			}
		} catch (Exception e) {
			e.printStackTrace();
			return e.getClass().getName();
		}
	}

	public String default_GererAnnonceAudioCatchExceptions(
			Boolean encodingActivated, String audioExtension, Integer bitrate,
			String conversionBinaryName, Boolean encodingSuccess,
			UC_DestAudioFile destAudioFile) {
		try {
			TemporaryFolder rootFolder = setup();
			try {
				GererAnnonceAudio aa;
				aa = gererAnnonceAudio_DefaultCase(rootFolder);

				final File downloadDir = rootFolder.newFolder();
				return allGererAnnonceAudio(encodingActivated, audioExtension,
						bitrate, conversionBinaryName, encodingSuccess, aa,
						downloadDir, destAudioFile);
			} finally {
				teardown(rootFolder);

			}
		} catch (Exception e) {
			e.printStackTrace();
			return e.getClass().getName();
		}
	}

	public String downloadError_GererAnnonceAudio(Boolean encodingActivated,
			String audioExtension, Integer bitrate,
			String conversionBinaryName, Boolean encodingSuccess,
			GererAnnonceAudio aa, File downloadDir) throws Exception {

		downloadDir.mkdir();
		HttpDataObj httpDataObj = new HttpDataObj(downloadDir.getAbsolutePath()
				+ "/", "http://localhost/get");
		DataObject configAudioTmp = new AudioDataObject(encodingActivated,
				audioExtension); // url_embarque_server_get,

		PowerMockito.mockStatic(Mp3Encoder.class);
		final String destinationFullPath = downloadDir.getAbsolutePath() + "/"
				+ "10.151.156.180Mon_Nov_04_140724_CET_2013343.mp3";
		final String resourceName = TEST_MP3_FILENAME;

		Answer<Integer> encodingOperation = encodingSuccess ? successfullEncoding(
				resourceName, destinationFullPath) : encodingFailure();

		PowerMockito.when(Mp3Encoder.launchMp3Exec(anyString())).thenAnswer(
				encodingOperation);

		// When
		String flux = aa.demanderFlux(buildXmlRequest(audioExtension),
				configAudioTmp, httpDataObj);

		// Then
		return flux;
	}

	public String allGererAnnonceAudio(Boolean encodingActivated,
			String audioExtension, Integer bitrate,
			String conversionBinaryName, Boolean encodingSuccess,
			GererAnnonceAudio aa, File downloadDir,
			UC_DestAudioFile destAudioFile) throws Exception {

		downloadDir.mkdir();
		HttpDataObj httpDataObj = new HttpDataObj(downloadDir.getAbsolutePath()
				+ "/", "http://localhost/get");
		DataObject configAudioTmp = new AudioDataObject(encodingActivated,
				audioExtension); // url_embarque_server_get,

		final String destinationFullPath = downloadDir.getAbsolutePath() + "/"
				+ "10.151.156.180Mon_Nov_04_140724_CET_2013343.mp3";
		final String resourceName = TEST_MP3_FILENAME;

		Answer<Integer> encodingOperation = encodingSuccess ? successfullEncoding(
				resourceName, destinationFullPath) : encodingFailure();

		PowerMockito.mockStatic(Mp3Encoder.class);
		PowerMockito.when(Mp3Encoder.launchMp3Exec(anyString())).thenAnswer(
				encodingOperation);

		destAudioFile.activateUC(destinationFullPath);

		// When
		String flux = aa.demanderFlux(buildXmlRequest(audioExtension),
				configAudioTmp, httpDataObj);

		// Then
		return flux;
	}

	private Answer<Integer> successfullEncoding(final String sourceName,
			final String destinationFullPath) {
		return new Answer<Integer>() {

			public Integer answer(InvocationOnMock invocation) throws Throwable {
				String testFileName = getClass().getClassLoader()
						.getResource(sourceName).getFile();
				FileUtils.copyFile(new File(testFileName).getAbsolutePath(),
						destinationFullPath);
				return 0;
			}
		};
	}

	private Answer<Integer> encodingFailure() {
		return new Answer<Integer>() {

			public Integer answer(InvocationOnMock invocation) throws Throwable {
				return 1;
			}
		};
	}

	private void teardown(TemporaryFolder rootFolder) {
		rootFolder.delete();
	}

	private TemporaryFolder setup() throws IOException {
		TemporaryFolder rootFolder = new TemporaryFolder();
		rootFolder.create();
		return rootFolder;
	}

	private String buildXmlRequest(final String audioExtension)
			throws ObixTmlgExeption {
		TemplateEngine te = new MustacheTemplateEngine();
		@SuppressWarnings("serial")
		Map<String, String> vals = new HashMap<String, String>() {
			{
				put("SOURCEFORMAT", audioExtension);
			}
		};
		String message = te
				.compile(
						"src/test/resources/ogive2pscsAudio_with_transformed_audio.xml",
						vals);
		return message;
	}

	private GererAnnonceAudio gererAnnonceAudio_DefaultCase(
			final TemporaryFolder rootFolder) throws IOException {
		final File lclsrvrFolder = rootFolder.newFolder();
		lclsrvrFolder.mkdir();

		GererAnnonceAudio aa = new GererAnnonceAudio() {
			@Override
			public byte[] downloadAudioFileFromHttpServer(String fileUrl)
					throws CoreException {
				try {
					InputStream resource = getClass()
							.getClassLoader()
							.getResourceAsStream(
									"10.151.156.180Mon_Nov_04_140724_CET_2013343.wav");
					byte[] bts = new byte[1000000];
					resource.read(bts);
					return bts;
				} catch (Exception e) {
					throw new CoreException(e);
				}
			}

			@Override
			protected String getLocalServerFolder() {

				return lclsrvrFolder.getAbsolutePath();
			}
		};
		return aa;
	}

	private GererAnnonceAudio gererAnnonceAudio_DownloadError()
			throws IOException {

		GererAnnonceAudio aa = new GererAnnonceAudio() {
			@Override
			public byte[] downloadAudioFileFromHttpServer(String fileUrl)
					throws CoreException {
				throw new CoreException("cant download");
			}
		};
		return aa;
	}
}
