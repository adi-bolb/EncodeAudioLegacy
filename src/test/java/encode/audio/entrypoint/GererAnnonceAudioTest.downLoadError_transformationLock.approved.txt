[true, mp3, 128, lame, true] = encode.audio.entrypoint.AppTechnicalException 
[false, mp3, 128, lame, true] = encode.audio.entrypoint.AppTechnicalException 
[true, wav, 128, lame, true] = encode.audio.entrypoint.AppTechnicalException 
[false, wav, 128, lame, true] = encode.audio.entrypoint.AppTechnicalException 
[true, mp3, 128, lame, false] = encode.audio.entrypoint.AppTechnicalException 
[false, mp3, 128, lame, false] = encode.audio.entrypoint.AppTechnicalException 
[true, wav, 128, lame, false] = encode.audio.entrypoint.AppTechnicalException 
[false, wav, 128, lame, false] = encode.audio.entrypoint.AppTechnicalException 
